p2p/protocol/identify/id.go

```diff
@@ -395,28 +399,35 @@ func (ids *IDService) sendIdentifyResp(s network.Stream) {
 	}
 
 	ph.snapshotMu.RLock()
-	mes := &pb.Identify{}
-	ids.populateMessage(mes, c, ph.snapshot)
-	w := ggio.NewDelimitedWriter(s)
-	w.WriteMsg(mes)
-
+	ids.writeChunkedIdentifyMsg(c, ph.snapshot, s)
 	log.Debugf("%s sent message to %s %s", ID, c.RemotePeer(), c.RemoteMultiaddr())
 }
 
 func (ids *IDService) handleIdentifyResponse(s network.Stream) {
 	c := s.Conn()
 
-	r := ggio.NewDelimitedReader(s, 2048)
+	r := ggio.NewDelimitedReader(s, legacyIDSize)
```

Comment:
This is weird. You should not create one delimited reader per message. One is enough, with buffer size `signedIDSize`. The second argument is the _capacity_ of the buffer.